src/runtime_src/core/pcie/windows/shim.cpp

```diff
@@ -1610,3 +1610,387 @@ xclReadTraceData(xclDeviceHandle handle, void* traceBuf, uint32_t traceBufSz,
+      samResult->CuExecCycles[s]     = ((uint64_t)(temp[1])) << 32;
+      samResult->CuStallExtCycles[s] = ((uint64_t)(temp[2])) << 32;
+      samResult->CuStallIntCycles[s] = ((uint64_t)(temp[3])) << 32;
+      samResult->CuStallStrCycles[s] = ((uint64_t)(temp[4])) << 32;
+      samResult->CuMinExecCycles[s]  = ((uint64_t)(temp[5])) << 32;
+      samResult->CuMaxExecCycles[s]  = ((uint64_t)(temp[6])) << 32;
+      samResult->CuStartCount[s]     = ((uint64_t)(temp[7])) << 32;
+
+      if(hasDataflow) {
+        uint64_t dfTmp[2] = {0};
+        shim->read(XCL_ADDR_SPACE_DEVICE_PERFMON, baseAddress[s] + XAM_BUSY_CYCLES_UPPER_OFFSET, &dfTmp[0], 4);
+        shim->read(XCL_ADDR_SPACE_DEVICE_PERFMON, baseAddress[s] + XAM_MAX_PARALLEL_ITER_UPPER_OFFSET, &dfTmp[1], 4);
+        samResult->CuBusyCycles[s]      = dfTmp[0] << 32;
+        samResult->CuMaxParallelIter[s] = dfTmp[1] << 32;
+      }
+    }
+
+    for (int c=0; c < XAM_DEBUG_SAMPLE_COUNTERS_PER_SLOT; c++)
+      size += shim->read(XCL_ADDR_SPACE_DEVICE_PERFMON, baseAddress[s]+sam_offsets[c], &temp[c], 4);
+
+    samResult->CuExecCount[s]      |= temp[0];
+    samResult->CuExecCycles[s]     |= temp[1];
+    samResult->CuStallExtCycles[s] |= temp[2];
+    samResult->CuStallIntCycles[s] |= temp[3];
+    samResult->CuStallStrCycles[s] |= temp[4];
+    samResult->CuMinExecCycles[s]  |= temp[5];
+    samResult->CuMaxExecCycles[s]  |= temp[6];
+    samResult->CuStartCount[s]     |= temp[7];
+
+    if(hasDataflow) {
+      uint64_t dfTmp[2] = {0};
+      shim->read(XCL_ADDR_SPACE_DEVICE_PERFMON, baseAddress[s] + XAM_BUSY_CYCLES_OFFSET, &dfTmp[0], 4);
+      shim->read(XCL_ADDR_SPACE_DEVICE_PERFMON, baseAddress[s] + XAM_MAX_PARALLEL_ITER_OFFSET, &dfTmp[1], 4);
+
+      samResult->CuBusyCycles[s]      |= dfTmp[0] << 32;
+      samResult->CuMaxParallelIter[s] |= dfTmp[1] << 32;
+    } else {
+      samResult->CuBusyCycles[s]      = samResult->CuExecCycles[s];
+      samResult->CuMaxParallelIter[s] = 1;
+    }
+  }
+  return size;
+}
+
+
+size_t xclDebugReadIPStatus(xclDeviceHandle handle, xclDebugReadType type, void* debugResults)
+{
+  xrt_core::message::send(xrt_core::message::severity_level::XRT_DEBUG, "XRT", "xclDebugReadIPStatus()");
+  switch (type) {
+    case XCL_DEBUG_READ_TYPE_LAPC :
+      return xclDebugReadCheckers(handle, reinterpret_cast<xclDebugCheckersResults*>(debugResults));
+    case XCL_DEBUG_READ_TYPE_AIM :
+      return xclDebugReadCounters(handle, reinterpret_cast<xclDebugCountersResults*>(debugResults));
+    case XCL_DEBUG_READ_TYPE_AM :
+      return xclDebugReadAccelMonitorCounters(handle, reinterpret_cast<xclAccelMonitorCounterResults*>(debugResults));
+    case XCL_DEBUG_READ_TYPE_ASM :
+      return xclDebugReadStreamingCounters(handle, reinterpret_cast<xclStreamingDebugCountersResults*>(debugResults));
+    case XCL_DEBUG_READ_TYPE_SPC:
+      return xclDebugReadStreamingCheckers(handle, reinterpret_cast<xclDebugStreamingCheckersResults*>(debugResults));
+    default:
```

Comment:
We have the compiler warnings turn all the way up.  So if an enumeration type is missing, and compiler error will be produced.  As such, the default option is no longer needed.